test/integration/ProjectBrowser.kt

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+// We need to access private function Studio.MainWindowColumn, this allows us to.
+// Do not use this outside of tests anywhere. It is extremely dangerous to do so.
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package com.vaticle.typedb.studio.test.integration
+
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import com.vaticle.typedb.studio.Studio
+import com.vaticle.typedb.studio.framework.common.WindowContext
+import com.vaticle.typedb.studio.state.StudioState
+import kotlinx.coroutines.delay
+import org.junit.Rule
+import org.junit.Test
+
+class ProjectBrowser {
+    @get:Rule
+    val composeRule = createComposeRule()
+
+    @Test
+    fun `Create a Directory`() {
+        val funcName = object{}.javaClass.enclosingMethod.name
```

Comment:
Sometimes we set up a test environment and copy some of the provided files under ./data into it so that the test can modify its environment without disturbing other tests or having to restore the data to its previous state.

Writing out the test name as a string within the function felt wrong, but this syntax is a little ugly.

Reply:
Ugly syntax is fine, but we should scope it by introducing a helper function. Something like...
```kt
fun funcName(obj: Any) = obj.javaClass.enclosingMethod.name
```
What are we using `funcName` for precisely?

Reply:
Let's generate a UUID for each test and use that in place of `funcName` - and also in place of `DB_NAME`.

Reply:
This has been done.